integration-cli/docker_cli_run_test.go

```diff
@@ -1298,16 +1260,22 @@ func (s *DockerCLIRunSuite) TestRunDNSOptions(c *testing.T) {
 
 	actual := regexp.MustCompile("(?m)^#.*$").ReplaceAllString(result.Stdout(), "")
 	actual = strings.ReplaceAll(strings.Trim(actual, "\r\n"), "\n", " ")
-	if actual != "nameserver 127.0.0.1 search mydomain options ndots:9" {
-		c.Fatalf("nameserver 127.0.0.1 expected 'search mydomain options ndots:9', but says: %q", actual)
+	if actual != "nameserver 127.0.0.11 search mydomain options ndots:9" {
+		c.Fatalf("nameserver 127.0.0.11 expected 'search mydomain options ndots:9', but says: %q", actual)
+	}
+	if !strings.Contains(result.Stdout(), "ExtServers: [127.0.0.1]") {
+		c.Fatalf("expected 'ExtServers: [127.0.0.1]' was not found in %q", result.Stdout())
 	}
 
 	out := cli.DockerCmd(c, "run", "--dns=1.1.1.1", "--dns-search=.", "--dns-opt=ndots:3", "busybox", "cat", "/etc/resolv.conf").Combined()
 
 	actual = regexp.MustCompile("(?m)^#.*$").ReplaceAllString(out, "")
 	actual = strings.ReplaceAll(strings.Trim(strings.Trim(actual, "\r\n"), " "), "\n", " ")
-	if actual != "nameserver 1.1.1.1 options ndots:3" {
-		c.Fatalf("expected 'nameserver 1.1.1.1 options ndots:3', but says: %q", actual)
+	if actual != "nameserver 127.0.0.11 options ndots:3" {
+		c.Fatalf("expected 'nameserver 127.0.0.11 options ndots:3', but says: %q", actual)
+	}
+	if !strings.Contains(out, "ExtServers: [1.1.1.1]") {
```

Comment:
Nothing urgent, but I saw the `1.1.1.1` and for a second thought we still set some default somewhere; but it looks like this is the IP we set as part of the test.

It made me thinking though; should we start practicing "best practice" and use designated example IP-addresses for these; similar to how we use designated example domains? See https://datatracker.ietf.org/doc/html/rfc5737

Reply:
Yes, a tidy-up of this and other old tests would be good. (I don't think it needs to be tied to this change though.)

Reply:
For sure; nothing urgent! My curiosity here was if we depended on those resolvers to make the test work (and in that case possibly having CI instability if those resolvers would be unreachable). That doesn't appear to be the case here (we only check the content of the file), but I guess having "designated" IP-addresses could possibly help taking away that confusion

LOL, that said; it's much easier to recognise "example.com" than one of those IP-addresses (we could have a `const` for them but that may be a bit too much.